File: cpp/002.cpp
#include <iostream>
using namespace std;

int main() {
	int a = 1, b = 1;
	
	int sum = 0;
	while (a < 4000000) {
		if(a % 2 == 0) {
			sum += a;
		}
		int c = a + b;
		a = b;
		b = c;
	}
	cout << sum << endl;
}

File: rs/002.rs
fn even(n: i32) -> bool {
    n % 2 == 0
}

fn main() {
    let mut a = 1;
    let mut b = 1;
    let mut sum = 0;

    while a <= 4_000_000 {
        if even(a) {
            sum += a;
        }

        let c = a + b;
        a = b;
        b = c;
    }

    println!("{}", sum);
}
